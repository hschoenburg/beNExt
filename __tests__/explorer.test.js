/* eslint-env jest */

import { shallow, mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import Explorer from '../pages/explorer/index.js'

let testProps = {
  type: 'height',
  term: 444,
  history: []
}

describe('Initial Render', () => {
  it('correctly renders when passed type+term props', () => {
    const wrap = mount(<Explorer {...testProps} />)
    expect(wrap.find('.focus').childAt(0).text()).toEqual('Height: 444')
  })
})

describe('focusExplorer', () => {
  it.only('should not update history if type+term are already in history', () => {
    let subject = mount(<Explorer {...testProps} />)
    subject.instance().focusExplorer(null, testProps)
    let history = subject.state('history').slice()
    subject.instance().focusExplorer(null, testProps) // run twice 
    let newHistory = subject.state('history').slice()
    expect(newHistory).toEqual(history)
  })

  it('should update state if type+term are new', () => {
    let newProps = {
      type: 'address',
      term: '0xsjhdfbsjhb'
    }
    let subject = mount(<Explorer {...testProps} />)
    let history = subject.state('history').slice()
    subject.instance().focusExplorer(null, newProps)
    let newHistory = subject.state().history

    expect(newHistory).not.toEqual(history)
    expect(newHistory[0].type).toEqual('height')
    expect(subject.state().type).toEqual('address')
  })
})

describe('shallow Home render', () => {
  it('Home shows 3 rows', () => {
    const wrap = shallow(<Explorer />)
    expect(wrap.find('div.row').length).toBe(3)
    expect(wrap.find('div.hero').length).toBe(1)
    // expect(wrap.find(Link).length).toBe(3)
  })
})

describe('Index snapshot', () => {
  it('Explorer shows 3 rows', () => {
    const component = renderer.create(<Explorer />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
