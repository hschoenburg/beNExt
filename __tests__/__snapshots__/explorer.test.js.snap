// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index snapshot Explorer shows 3 rows 1`] = `
<div
  className="jsx-3933479298"
>
  <nav
    className="jsx-2055897931"
  >
    <ul
      className="jsx-2055897931"
    >
      <li
        className="jsx-2055897931"
      >
        <a
          className="jsx-2055897931"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <ul
        className="jsx-2055897931"
      >
        <li
          className="jsx-2055897931"
        >
          <a
            className="jsx-2055897931"
            href="/explorer"
            onClick={[Function]}
          >
            Explorer
          </a>
        </li>
        <li
          className="jsx-2055897931"
        >
          <a
            className="jsx-2055897931"
            href="/market"
            onClick={[Function]}
          >
            Market
          </a>
        </li>
      </ul>
    </ul>
  </nav>
  <div
    className="jsx-3933479298 hero"
  >
    <h1
      className="jsx-3933479298 title"
    >
      Bcoin Explorer
    </h1>
    <p
      className="jsx-3933479298 description"
    >
      Make Crypto Useful
    </p>
  </div>
  <div
    className="jsx-3933479298 row"
  >
    <div
      className="jsx-266566849 explorer-search"
    >
      <form
        className="jsx-266566849"
        onSubmit={[Function]}
      >
        <input
          className="jsx-266566849"
          id="ExplorerSearch"
          onChange={[Function]}
          type="text"
        />
        <input
          className="jsx-266566849"
          type="submit"
          value="Search"
        />
      </form>
    </div>
  </div>
  <div
    className="jsx-3933479298 row"
  >
    <h3>
      Default Focus State
    </h3>
  </div>
  <div
    className="jsx-3933479298 row"
  />
</div>
`;
